algorithms/src/encryption/ecies_poseidon.rs

```diff
@@ -117,7 +117,26 @@ where
     }
 
     fn generate_private_key<R: Rng + CryptoRng>(&self, rng: &mut R) -> <Self as EncryptionScheme>::PrivateKey {
-        Self::PrivateKey::rand(rng)
+        // Keep trying until finding a key that is within the field's capacity bit limits.
+        loop {
+            let key = Self::PrivateKey::rand(rng);
+            let bits = key.to_bits_le();
+
+            let mut flag = false;
+            for bit in bits
+                .iter()
+                .skip(<TE::ScalarField as PrimeField>::Parameters::CAPACITY as usize)
+            {
+                if *bit == true {
+                    flag = true;
+                    break;
+                }
+            }
+
+            if flag == false {
```

Comment:
```suggestion
            if !flag {
```
(even if `clippy` doesn't complain about it yet, it will in the future, as it's a `bool` literal comparison)